card/mod/05_standard/set/right/account.rb

```diff
@@ -12,36 +12,21 @@ def blocked?;  status == 'blocked' end
 def built_in?; status == 'system'  end
 def pending?;  status == 'pending' end
 
-def authenticate_by_token val
+def validate_token! test_token
   tcard = token_card
-  error = token_error(tcard, val)
-  if error == :none
-    Auth.as_bot { tcard.delete! }
-    left.id
-  else
-    error
-  end
-end
-
-def token_error tcard, val
-  case
-  when !tcard
-    :token_not_found
-  when token != val
-    :incorrect_token
-  when tcard.updated_at <= Card.config.token_expiry.ago
-    # < means "before"
-    :token_expired
-  when !left || !left.accountable?
-    :illegal_account
-  else
-    :none
+  tcard.validate! test_token
+  copy_errors tcard
+  if errors.empty?
```

Comment:
Use a guard clause instead of wrapping the code inside a conditional expression.